File: umbrella-insurance-react-ui/src/models/teams/v1/Team.ts
import { GameType } from "../../gameTypes/v1/GameType";
import { LevelOfCompetition } from "../../levelOfCompetitions/v1/LevelOfCompetition";
import { Season } from "../../seasons/v1/Season";
import { Location } from "../../geographies/locations/v1/Location";
import { TeamType } from "../teamTypes/v1/TeamType";

export class Team {
    id?: number;
    teamName?: string;
    teamType?: TeamType;
    location?: Location;
    firstSeason?: Season;
    lastSeason?: Season;
    logoName?: string;
    logoImage?: number[];
    levelOfCompetition?: LevelOfCompetition;
    gameType?: GameType;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.teamName) {
                this.teamName = obj.teamName;
            }
            if(obj?.teamType) {
                this.teamType = new TeamType(obj.teamType);
            }
            if(obj?.location) {
                this.location = new Location(obj.location);
            }
            if(obj?.firstSeason) {
                this.firstSeason = new Season(obj.firstSeason);
            }
            if(obj?.lastSeason) {
                this.lastSeason = new Season(obj.lastSeason);
            }
            if(obj?.logoName) {
                this.logoName = obj.logoName;
            }
            if(obj?.logoImage) {
                this.logoImage = obj.logoImage;
            }
            if(obj?.levelOfCompetition) {
                this.levelOfCompetition = new LevelOfCompetition(obj.levelOfCompetition);
            }
            if(obj?.gameType) {
                this.gameType = new GameType(obj.gameType);
            }
        }
    }
}

File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Team.java
package com.umbrella.insurance.core.models.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "teams", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "teams_un", columnNames = {"team_name", "level_of_competition_id", "game_type_id"})
})
public class Team {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "team_id", nullable = false)
    private Long id;

    @NotNull
    @Column(name = "team_name", nullable = false, length = Integer.MAX_VALUE)
    private String teamName;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "team_type_id")
    private TeamType teamType;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "location_id")
    private Location location;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "first_season_id")
    private Season firstSeason;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "last_season_id")
    private Season lastSeason;

    @Column(name = "logo_name", length = Integer.MAX_VALUE)
    private String logoName;

    @Column(name = "logo_image")
    private byte[] logoImage;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "level_of_competition_id")
    private LevelOfCompetition levelOfCompetition;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "game_type_id")
    private GameType gameType;

}